clientconn.go

```diff
@@ -924,12 +918,18 @@ func (ac *addrConn) connect() error {
 	return nil
 }
 
+func equalAddress(a, b resolver.Address) bool {
```

Comment:
Probably makes sense to add a function level comment here saying why we are not using the `resolver.Address.Equal` and instead defining our own. Something as simple as:

```
// equalAddress returns true is a and b are considered equal.
// This is different from the Equal method on the resolver.Address type 
// which considers all fields to determine equality. Here, we only consider
// fields that are meaningful to the subConn.
func equalAddress(a, b resolver.Address) bool { ... }
```

Reply:
Done.

Reply:
Renaming the function might be an even nicer improvement.  `equalIgnoringBalAttributes`?

Reply:
Done.